pkg/scaling/resolver/gcp_secretmanager_handler_test.go

```diff
@@ -0,0 +1,135 @@
+	logf "sigs.k8s.io/controller-runtime/pkg/log"
+
+	kedav1alpha1 "github.com/kedacore/keda/v2/apis/keda/v1alpha1"
+)
+
+type GCPHandlerTestCase struct {
+	Name             string
+	Config           *kedav1alpha1.GCPSecretManager
+	TriggerNamespace string
+	ExpectedError    string
+}
+
+func TestGCPSecretManagerHandlerInitialize(t *testing.T) {
+	testCases := []GCPHandlerTestCase{
+		{
+			Name: "Invalid Pod identity provider",
+			Config: &kedav1alpha1.GCPSecretManager{
+				PodIdentity: &kedav1alpha1.AuthPodIdentity{
+					Provider: "xyz",
+				},
+			},
+			TriggerNamespace: "testNamespace",
+			ExpectedError:    "gcp secret manager does not support pod identity provider - xyz",
+		},
+		{
+			Name: "Missing credentials and pod identity provider",
+			Config: &kedav1alpha1.GCPSecretManager{
+				Credentials: nil,
+				PodIdentity: &kedav1alpha1.AuthPodIdentity{
+					Provider: "",
+				},
+			},
+			TriggerNamespace: "testNamespace",
+			ExpectedError:    "clientSecret is expected when not using a pod identity provider",
+		},
+		{
+			Name: "Empty trigger namespace",
+			Config: &kedav1alpha1.GCPSecretManager{
+				Credentials: &kedav1alpha1.GCPCredentials{
+					ClientSecret: kedav1alpha1.GCPSecretmanagerClientSecret{
+						ValueFrom: kedav1alpha1.ValueFromSecret{
+							SecretKeyRef: kedav1alpha1.SecretKeyRef{
+								Name: "testSecretName",
+								Key:  "testSecretKey",
+							},
+						},
+					},
+				},
+				PodIdentity: &kedav1alpha1.AuthPodIdentity{
+					Provider: kedav1alpha1.PodIdentityProviderNone,
+				},
+			},
+			TriggerNamespace: "",
+			ExpectedError:    "clientSecret is expected when not using a pod identity provider",
+		},
+		{
+			Name: "Empty credentials secret name",
+			Config: &kedav1alpha1.GCPSecretManager{
+				Credentials: &kedav1alpha1.GCPCredentials{
+					ClientSecret: kedav1alpha1.GCPSecretmanagerClientSecret{
+						ValueFrom: kedav1alpha1.ValueFromSecret{
+							SecretKeyRef: kedav1alpha1.SecretKeyRef{
+								Name: "",
+								Key:  "testSecretKey",
+							},
+						},
+					},
+				},
+				PodIdentity: &kedav1alpha1.AuthPodIdentity{
+					Provider: kedav1alpha1.PodIdentityProviderNone,
+				},
+			},
+			TriggerNamespace: "testNamespace",
+			ExpectedError:    "clientSecret is expected when not using a pod identity provider",
+		},
+		{
+			Name: "Empty credentials secret key",
+			Config: &kedav1alpha1.GCPSecretManager{
+				Credentials: &kedav1alpha1.GCPCredentials{
+					ClientSecret: kedav1alpha1.GCPSecretmanagerClientSecret{
+						ValueFrom: kedav1alpha1.ValueFromSecret{
+							SecretKeyRef: kedav1alpha1.SecretKeyRef{
+								Name: "testSecretName",
+								Key:  "",
+							},
+						},
+					},
+				},
+				PodIdentity: &kedav1alpha1.AuthPodIdentity{
+					Provider: kedav1alpha1.PodIdentityProviderNone,
+				},
+			},
+			TriggerNamespace: "testNamespace",
+			ExpectedError:    "clientSecret is expected when not using a pod identity provider",
+		},
```

Comment:
can you please add two happy cases too?

1) for correct workload identity provider
```go
			Name: "Using workload identity",
￼			Config: &kedav1alpha1.GCPSecretManager{
￼				PodIdentity: &kedav1alpha1.AuthPodIdentity{
￼					Provider: kedav1alpha1.PodIdentityProviderGCP,
￼				},
￼			},
￼			TriggerNamespace: "testNamespace",
            ExpectedError: "",
```
2) and for correct client secret
```go
			Name: "Empty credentials secret key",
￼			Config: &kedav1alpha1.GCPSecretManager{
￼				Credentials: &kedav1alpha1.GCPCredentials{
￼					ClientSecret: kedav1alpha1.GCPSecretmanagerClientSecret{
￼						ValueFrom: kedav1alpha1.ValueFromSecret{
￼							SecretKeyRef: kedav1alpha1.SecretKeyRef{
￼								Name: "testSecretName",
                                Namespace: "testNamespace",
￼								Key:  "key",
￼							},
￼						},
￼					},
￼				},
￼			},
￼			TriggerNamespace: "testNamespace",
￼			ExpectedError:    "",
```